pygeoapi/provider/mongo.py

```diff
@@ -76,21 +75,25 @@ def get_fields(self):
         :returns: dict of fields
         """
 
-        map = Code(
-            "function() { for (var key in this.properties) "
-            "{ emit(key, null); } }")
-        reduce = Code("function(key, stuff) { return null; }")
-        result = self.featuredb[self.collection].map_reduce(
-            map, reduce, "myresults")
+        pipeline = [
```

Comment:
map_reduce is not supported in pymongo 4.6.3, so we need to migrate this code to an aggregation pipeline

https://www.mongodb.com/docs/manual/reference/map-reduce-to-aggregation-pipeline/